packages/airnode-adapter/src/response-processing/extraction.test.ts

```diff
@@ -66,11 +64,95 @@ describe('extractAndEncodeValue', () => {
-      value: '75051',
+      rawValue: data,
+      values: ['75051'],
       encodedValue: '0x000000000000000000000000000000000000000000000000000000000001252b',
     });
   });
+
+  it('accepts "" (empty string) for _times parameter', () => {
+    const data = { a: { b: [{ c: 1 }, { d: '750.51' }] } };
+    const parameters: ReservedParameters = { _path: 'a.b.1.d', _type: 'int256', _times: '' };
+    const res = extractAndEncodeResponse(data, parameters);
+    expect(res).toEqual({
+      rawValue: data,
+      values: ['750'],
+      encodedValue: '0x00000000000000000000000000000000000000000000000000000000000002ee',
+    });
+  });
+
+  it('empty string in path returns the whole API response', () => {
+    const data = 123456;
+    const parameters: ReservedParameters = { _path: '', _type: 'int256' };
+    const res = extractAndEncodeResponse(data, parameters);
+    expect(res).toEqual({
+      rawValue: 123456,
+      values: ['123456'],
+      encodedValue: '0x000000000000000000000000000000000000000000000000000000000001e240',
+    });
+  });
+});
+
+describe('extract and encode multiple values', () => {
+  it('works for basic request', () => {
+    const data = { a: { b: [{ c: 1 }, { d: '750.51' }] } };
+    const parameters: ReservedParameters = { _path: 'a.b.1.d,a.b.0.c', _type: 'int256,bool', _times: '100,' };
+    const res = extractAndEncodeResponse(data, parameters);
+    expect(res).toEqual({
+      rawValue: data,
+      values: ['75051', true],
+      encodedValue:
+        '0x000000000000000000000000000000000000000000000000000000000001252b0000000000000000000000000000000000000000000000000000000000000001',
+    });
+  });
+
+  it('works for more complex request', () => {
+    const data = [12.3, 45.6];
+    const parameters: ReservedParameters = { _path: ',0', _type: 'int256[],int256', _times: '100,1000' };
+    const res = extractAndEncodeResponse(data, parameters);
+    expect(res).toEqual({
+      rawValue: data,
+      values: [['1230', '4560'], '12300'],
+      encodedValue:
+        '0x0000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000300c000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000004ce00000000000000000000000000000000000000000000000000000000000011d0',
+    });
+  });
+
+  it('correctly splits reserved parameters containing multiple values', () => {
+    expect(splitReservedParameters({ _type: 'uint256' })).toEqual([{ _type: 'uint256' }]);
+    expect(splitReservedParameters({ _type: 'uint256,string', _path: 'key,anotherKey' })).toEqual([
+      { _type: 'uint256', _path: 'key' },
+      { _type: 'string', _path: 'anotherKey' },
+    ]);
+    expect(splitReservedParameters({ _type: 'uint256, invalid' })).toEqual([
+      { _type: 'uint256' },
+      { _type: ' invalid' },
+    ]);
+    expect(splitReservedParameters({ _type: 'uint256,,,invalid' })).toEqual([
```

Comment:
FYI: This type is ofc invalid, but I want to double check that the function works correctly.